src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -281,18 +248,58 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
                     }
                     else
                     {
+                        // Prevent use of unassigned variables error
                         return;
                     }
 
-                    context.ReportDiagnostic(
-                        invocation.CreateDiagnostic(rule, ruleMessageMethod, ruleMessagePreferredMethod));
+                    context.ReportDiagnostic(awaitOperation.CreateDiagnostic(rule, ruleMessageMethod, ruleMessagePreferredMethod));
+                }
+            },
+            OperationKind.Await);
+        }
+
+        private static bool ShouldAnalyze(
+            IAwaitOperation awaitOperation,
+            IMethodSymbol configureAwaitMethod,
+            IMethodSymbol genericConfigureAwaitMethod,
+            INamedTypeSymbol streamType,
+            out IMethodSymbol? actualMethod)
+        {
+            actualMethod = null;
+
+            if (!awaitOperation.Children.Any())
```

Comment:
Please avoid using `Children` property. That is primarily meant for navigation of entire operation tree when the structure/kinds of nodes is not known. We should instead just check `if (!(awaitOperation.Operation is IInvocationOperation invocation))`

Reply:
Is there another way to retrieve the child `ReadAsync/WriteAsync` invocation when the first operation is the await?

Reply:
Nevermind. Based on the next comment, if I have to revert the changes from the last commit, I assume I don't have to address this.

Reply:
I believe it's `awaitOperation.Operation`

Reply:
I clarified the below comment with Manish. I don't need to revert the whole git commit. Instead, I should use awaitOperation.Operation directly as a substitute for traversing the Children or for creating the diagnostic.

Reply:
Thanks @bartonjs. GitHub did not show your message until I refreshed the page.